pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java

```diff
@@ -585,6 +585,15 @@ public static class Server {
     public static final String CONFIG_OF_STARTUP_ENABLE_SERVICE_STATUS_CHECK =
         "pinot.server.startup.enableServiceStatusCheck";
     public static final boolean DEFAULT_STARTUP_ENABLE_SERVICE_STATUS_CHECK = true;
+    // The timeouts above determine how long servers will poll their status before giving up.
+    // This configuration determines what we do when we give up. By default, we will mark the
+    // server as healthy and start the query server. If this is set to true, we instead throw
+    // an exception and exit the server. This is useful if you want to ensure that the server
+    // is always fully ready before accepting queries. But note that this can cause the server
+    // to never be healthy if there is some reason that it can never reach a GOOD status.
+    public static final String CONFIG_OF_EXIT_SERVER_ON_INCOMPLETE_STARTUP =
+        "pinot.server.starter.exitServerOnStartupStatusFailure";
+    public static final boolean DEFAULT_EXIT_SERVER_ON_INCOMPLETE_STARTUP = false;
```

Comment:
(optional) Suggest making it more specific and matching `CONFIG_OF_STARTUP_ENABLE_SERVICE_STATUS_CHECK`
```suggestion
    public static final String CONFIG_OF_STARTUP_EXIT_ON_SERVICE_STATUS_CHECK_FAILURE =
        "pinot.server.startup.exitOnServiceStatusCheckFailure";
    public static final boolean DEFAULT_STARTUP_EXIT_ON_SERVICE_STATUS_CHECK_FAILURE = false;
```

Reply:
apologies, missed this comment both times